### PR TITLE
test: dns record check integration tests

### PR DESCRIPTION
**Pull request by austinborn**
## Description

DNS Record Check integration tests. Note: source code changes were made to ensure the adapter works properly. If this seems wrong, I'm happy to discuss alternatives.

## Changes

- DNS Record Check integration tests
- Add `result` object to DNS Query adapter's response so DNS Record Check can use it
- Use the `name` attribute of Answer records to compare, not `record` attribute

## Steps to Test

`yarn && yarn setup && yarn test dns-record-check/test/integration`

## Quality Assurance

- [x] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `<ADAPTER_PACKAGE>/schemas/env.json` and `<ADAPTER_PACKAGE>/README.md`
- [x] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [x] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [x] This is related to a maximum of one Clubhouse/Shortcut story or GitHub issue
- [x] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)

### CODE REVIEW

**Review by austinborn — COMMENTED** [review]
Note to self: update package version and CHANGELOG

**Review by boxhock — APPROVED** [review]

File: .changeset/rude-bobcats-smile.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@chainlink/dns-record-check-adapter': patch
+---
+
+Integration tests for dns-record-check
```
Comment (austinborn) [inline .changeset/rude-bobcats-smile.md:5]:
```suggestion
dns-record-check src export change
```

File: .changeset/happy-penguins-act.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@chainlink/dns-record-check-adapter': minor
+---
+
+Add integration tests to dns-record-check. Add alias for `record` input param.
```
Comment (austinborn) [inline .changeset/happy-penguins-act.md:5]:
```suggestion
Add alias for `record` input param.
```

File: packages/composites/dns-record-check/README.md [hunk 3]
```diff
@@ -12,9 +12,9 @@ See the [Composite Adapter README](../README.md) for more information on how to
 
 ### Input Params
 
-| Required? |   Name   |             Description              | Options | Defaults to |
-| :-------: | :------: | :----------------------------------: | :-----: | :---------: |
-|    ✅     | `record` | name of record, eg: "adex-publisher" |         |             |
+| Required? |       Name       |             Description              | Options | Defaults to |
+| :-------: | :--------------: | :----------------------------------: | :-----: | :---------: |
+|    ✅     | `name`, `record` | name of record, eg: "adex-publisher" |         |             |
```
Comment (austinborn) [inline packages/composites/dns-record-check/README.md:17]:
```suggestion
| Required? |  Name  |                                      Description                                       | Options | Defaults to |
| :-------: | :----: | :------------------------------------------------------------------------------------: | :-----: | :---------: |
|    ✅     | `name`, `record` |                             name of record, eg: "adex-publisher"                                                      |         |             |
|    ✅     | `type`.                   |                 Query Type (either a numeric value or text), eg. "TXT"                                     |         |             |
|           | `do`                      | DO bit - set if client wants DNSSEC data (either boolean or numeric value), eg. "true" |         |             |
|           |  `cd`                     |   CD bit - set to disable validation (either boolean or numeric value), eg. "false"            |         |             |
```

**Comment by justinkaseman** [comment]
What is the status on this one?

**Comment by austinborn** [comment]
> What is the status on this one?

Let me try updating/fixing this one rn. I don't recall why we couldn't merge it

**Review by justinkaseman — APPROVED** [review]

**Pull request merged**